### PR TITLE
3006: Delete Registrations

### PR DESCRIPTION
**Pull request by McMutton**
Adds: DELETE /v2/registrations/<registration-id>

I have a question:

What should be the proper HTTP response code for successful DELETE requests? When I check, I see that for some cases we send `204 No Content` and for some cases it is `200 OK`. 

For this PR, I left it as `200 OK`. Please let me know if I should change it to `204 No Content`.

### CODE REVIEW

**Comment by kzangeli** [comment]
204 sounds good to me.
@jmcanterafonseca  and @fgalan should give us their opinion as well

**Comment by fgalan** [comment]
According to http://telefonicaid.github.io/fiware-orion/api/v2/latest/, it must be 204 No Content

File: test/functionalTest/cases/3006_delete_registrations/delete_reg_using_v2_created_using_v2.test [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+Content-Length: 0
+Location: /v2/registrations/REGEX([0-9a-f\-]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. DELETE the registration using APIv2
+=======================================
+HTTP/1.1 200 OK
```
Comment (fgalan) [inline test/functionalTest/cases/3006_delete_registrations/delete_reg_using_v2_created_using_v2.test]: 200 OK -> 204 No Content

Comment (McMutton) [reply test/functionalTest/cases/3006_delete_registrations/delete_reg_using_v2_created_using_v2.test]: Fixed in 7d5e794

**Comment by fgalan** [comment]
LGTM

Passing the ball to @kzangeli for final LGTM

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**